src/Tests/PathautoLocaleTest.php

```diff
@@ -84,5 +93,92 @@ function testLanguageAliases() {
+    \Drupal::service('router.builder')->rebuild();
+    \Drupal::service('entity.definition_update_manager')->applyUpdates();
+
+    // Create a pattern for English articles.
+    $pattern = $this->createPattern('node', '/the-articles/[node:title]');
+    $pattern->addSelectionCondition([
+      'id' => 'entity_bundle:node',
+      'bundles' => ['article' => 'article'],
+      'negate' => FALSE,
+      'context_mapping' => ['node' => 'node'],
+    ]);
+    $language_mapping = 'node:langcode:language';
+    $pattern->addSelectionCondition([
+      'id' => 'language',
+      'langcodes' => ['en' => 'en'],
+      'negate' => FALSE,
+      'context_mapping' => ['language' => $language_mapping],
+    ]);
+    $new_definition = new ContextDefinition('language', 'Language');
+    $new_context = new Context($new_definition);
+    $pattern->addContext($language_mapping, $new_context);
+    $pattern->save();
+
+    // Create a pattern for French articles.
+    $pattern = $this->createPattern('node', '/les-articles/[node:title]');
+    $pattern->addSelectionCondition([
+      'id' => 'entity_bundle:node',
+      'bundles' => ['article' => 'article'],
+      'negate' => FALSE,
+      'context_mapping' => ['node' => 'node'],
+    ]);
+    $language_mapping = 'node:langcode:language';
+    $pattern->addSelectionCondition([
+      'id' => 'language',
+      'langcodes' => ['fr' => 'fr'],
+      'negate' => FALSE,
+      'context_mapping' => ['language' => $language_mapping],
+    ]);
+    $new_definition = new ContextDefinition('language', 'Language');
+    $new_context = new Context($new_definition);
+    $pattern->addContext($language_mapping, $new_context);
+    $pattern->save();
+
+    // Create a node and its translation. Assert aliases.
+    $edit = array(
+      'title[0][value]' => 'English node',
+    );
+    $this->drupalPostForm('node/add/article', $edit, t('Save and publish'));
+    $this->drupalGet('node/1/edit');
+    $english_node = $this->drupalGetNodeByTitle('English node');
+    $this->assertAlias('/node/' . $english_node->id(), '/the-articles/english-node', 'en');
+
+    $add_translation_url = Url::fromRoute('entity.node.content_translation_add', ['node' => $english_node->id(), 'source' => 'en', 'target' => 'fr']);
+    $edit = array(
+      'title[0][value]' => 'French node',
+    );
+    $this->drupalPostForm($add_translation_url, $edit, t('Save and keep published (this translation)'));
+    $this->rebuildContainer();
+    $english_node = $this->drupalGetNodeByTitle('English node');
+    $french_node = $english_node->getTranslation('fr');
+    $this->assertAlias('/node/' . $french_node->id(), '/les-articles/french-node', 'fr');
+
+    // Bulk delete and Bulk generate patterns. Assert aliases.
+    $this->deleteAllAliases();
+    // Bulk create aliases.
+    $edit = array(
+      'update[canonical_entities:node]' => TRUE,
+    );
+    $this->drupalPostForm('admin/config/search/path/update_bulk', $edit, t('Update'));
```

Comment:
not surprised that this doesn't work at all :)

We have to update EntityAliasTypeBase::bulkUpdate() to loop over all translations of an entity.